packages/camera/camera_avfoundation/ios/Classes/FLTCam.m

```diff
@@ -374,9 +374,8 @@ - (void)captureToFileWithCompletion:(void (^)(NSString *_Nullable,
     extension = @"jpg";
   }
 
-  AVCaptureFlashMode avFlashMode = FCPGetAVCaptureFlashModeForPigeonFlashMode(_flashMode);
-  if (avFlashMode != -1) {
-    [settings setFlashMode:avFlashMode];
+  if (_flashMode != FCPPlatformFlashModeTorch) {
```

Comment:
Is this because iOS doesn't support taking picture in torch mode? Maybe add a comment?

Reply:
It's because torch mode means the flash is forced to on all of the time even when not taking a picture, so there is no picture-level setting for it.

Reply:
Added a comment in the code.